frontends/ol-components/src/components/TiptapEditor/extensions/node/Image/ImageWithCaption.tsx

```diff
@@ -0,0 +1,216 @@
+import React from "react"
+import { NodeViewWrapper } from "@tiptap/react"
+import type { ReactNodeViewProps } from "@tiptap/react"
+import styled from "@emotion/styled"
+import NiceModal from "@ebay/nice-modal-react"
+import ImageAltTextInput from "./ImageAltTextInput"
+import { DefaultWidth, WideWidth, FullWidth } from "./Icons"
+
+const ARTICLE_MAX_WIDTH = 890
+const CONTAINER_PADDING = 24
+
+const Container = styled.div({
+  position: "relative",
+  margin: "2rem auto",
+  textAlign: "center",
+  width: "100%",
+
+  img: {
+    width: "100%",
+    height: "auto",
+    aspectRatio: "16/9",
+    borderRadius: "6px",
+    display: "block",
+  },
+
+  "&.layout-default img": {
+    width: "100%",
+  },
+
+  [`@media (min-width: ${ARTICLE_MAX_WIDTH + CONTAINER_PADDING * 2}px)`]: {
+    "&.layout-wide img": {
+      width: "90vw",
+      maxWidth: "90vw",
+      position: "relative",
+      left: "50%",
+      right: "50%",
+      marginLeft: "-45vw",
+      marginRight: "-45vw",
+    },
+  },
+
+  "&.layout-full img": {
+    width: "100vw",
+    maxWidth: "100vw",
+    position: "relative",
+    left: "50%",
+    right: "50%",
+    marginLeft: "-50vw",
+    marginRight: "-50vw",
+  },
+
+  ".caption-input": {
+    width: "100%",
+    border: "none",
+    outline: "none",
+    background: "transparent",
+    fontSize: "14px",
+    padding: "0",
+  },
+
+  ".media-layout-toolbar": {
+    display: "none",
+    position: "absolute",
+    top: "-34px",
+    left: "50%",
+    transform: "translateX(-50%)",
+    zIndex: "2000",
+    backgroundColor: "rgb(0 0 0 / 85%)",
+    padding: "6px 10px",
+    borderRadius: "8px",
+    gap: "8px",
+    width: "250px",
+    justifyContent: "center",
+
+    "&::after": {
+      content: '""',
+      position: "absolute",
+      top: "100%",
+      left: "50%",
+      transform: "translateX(-50%)",
+      width: 0,
+      height: 0,
+      borderLeft: "8px solid transparent",
+      borderRight: "8px solid transparent",
+      borderTop: "8px solid rgb(0 0 0 / 85%)",
+    },
+
+    button: {
+      width: "40px",
+      height: "28px",
+      borderRadius: "4px",
+      border: "none",
+      background: "transparent",
+      filter: "brightness(0.9)",
+      color: "white",
+      cursor: "pointer",
+
+      "&.active": {
+        background: "#9be19b",
+        color: "black",
+        fontWeight: "bold",
+      },
+    },
+
+    ".alt-text-button": {
+      color: "white",
+      fontSize: "12px",
+      borderRadius: "4px",
+      padding: "2px 6px",
+      width: "100px",
+    },
+  },
+
+  "&:hover": {
+    ".media-layout-toolbar": {
+      display: "flex",
+    },
+  },
+})
+
+enum Layout {
+  default = "default",
+  wide = "wide",
+  full = "full",
+}
+
+const Image = styled.img<{ layout: Layout }>(({ layout }) => ({
+  "&&": {
+    borderRadius: layout === Layout.full ? 0 : "8px",
+  },
+}))
+
+const Caption = styled.p(({ theme }) => ({
+  "&&&&&": {
```

Comment:
Seems like we are fighting with the vendored stylesheets.

Do you think they are generally useful, or should we remove them? Or should we keep the vendor discussion closed...